### DISCUSSION TITLE
How to tell the grabber to avoid ore if an alien is near it?

### DISCUSSION THREAD

**Discussion by Zemogiter** [colobot/colobot#1531] [Q&A] [answered]
I've tried stuff like this:
```
public point object::AyyCheck()
{
	object nearestAlien, alien;
	int alienList[], nr;
	nr = 0;
	
	alienList[nr++] = AlienWasp;
	alienList[nr++] = AlienAnt;
	alienList[nr++] = AlienSpider;
	alienList[nr++] = AlienWorm;
	alienList[nr++] = AlienEgg;
	alienList[nr++] = AlienQueen;
	
	for(nr = 0 ; nr < 6 ; nr++)
	{
		alien = radar(alienList[nr]);
		
	}
	return alien.position;
	
}
```
then:
`if (derrick != null && distance(ore.position, AyyCheck()) < 30)`
However it crashes the program with the "unknown object" error if there are no aliens on map. Can someone help me?

**Comment by tomangelo2** [comment, marked as answer]
radar() returns `null`, if there is no such object on the map. With that in mind, you can make it like that
```cpp
if(alien == null) //check if value found by radar exists
  return new point(9999,9999,999); //return some impossible value
else
  return alien.position; //return actual alien position
```

This way, if there is no alien on the map, your main function will think that closest alien is far, far away.

Another suggestion: before you assign new found alien to the alien variable, you probably should check if it's closer to your base that the one that was already assigned. Otherwise, the alien #.4 that's next to your base will be overridden by alien #.5, that's kilometers away.

**Reply by Zemogiter** [reply]
Thanks, works like a charm.
